com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java

```diff
@@ -79,8 +80,10 @@
                 }
             }
         } catch (IOException | IllegalConnectorArgumentsException e) {
-            return AdapterUtils.createAsyncErrorResponse(response, ErrorCode.ATTACH_FAILURE,
-                        String.format("Failed to attach to remote debuggee VM. Reason: %s", e.toString()));
+            throw new CompletionException(new DebugException(
+                String.format("Failed to attach to remote debuggee VM. Reason: %s", e.toString()),
```

Comment:
Many duplicate code for throw new CompletionException(new DebugException(xxxx)), make a utility to handle?

Reply:
Resolved. Use factory method to wrap DebugException as CompletionException.